### DISCUSSION TITLE
Better URL handling for private repo in the mobile app

### DISCUSSION THREAD

**Discussion by andi-bigswitch** [community/community#8575] [Mobile]
Situation/Context: I am logged into my account in the app. I am _not_ logged in in iOS Safari (and I would like to avoid logging in there).

When a coworker sends me a GitHub URL to a private repo to look at in chat, it opens in the browser. The browser shows the 404 page (expected). It does not offer to open the page on the app.

The app itself does not have a URL bar so I also can't manually copy and open the URL there.

The only way that I have found to open the URL in the app is to also sign into GitHub in Safari. Then Safari shows the suggestion to open the URL in the app.

Suggestion/Feature Request. Make it possible to open URLs inside the app without requiring the browser to be signed in. Two ideas:
1. On a Mobile browser, present the option to try opening in the app even when showing the 404 page
2. Make the Search bar on the app home section handle URLs.

**Comment by brianlovin** [comment]
@andi-bigswitch agreed that the browser/app authentication discrepancy is weird. In the short term, could you try going to your profile → settings → external links and enabling the "Clipboard Link Detection" toggle? That should help for now by presenting a toast that gives quick access to navigate to a GitHub URL when it's detected in the pasteboard.

**Reply by MathiasLui** [reply]
Still an issue. Getting invited to a private repo, clicking on it in the app, 404
